mlir/lib/Analysis/Presburger/IntegerRelation.cpp

```diff
@@ -1125,11 +1121,12 @@ unsigned IntegerRelation::gaussianEliminateVars(unsigned posStart,
 bool IntegerRelation::gaussianEliminate() {
   gcdTightenInequalities();
   unsigned firstVar = 0, vars = getNumVars();
-  unsigned nowDone, eqs, pivotRow;
+  unsigned nowDone, eqs;
+  std::optional<unsigned> pivotRow;
   for (nowDone = 0, eqs = getNumEqualities(); nowDone < eqs; ++nowDone) {
     // Finds the first non-empty column.
     for (; firstVar < vars; ++firstVar) {
-      if (!findConstraintWithNonZeroAt(firstVar, true, &pivotRow))
+      if (!(pivotRow = findConstraintWithNonZeroAt(firstVar, /*isEq=*/true)))
         continue;
       break;
```

Comment:
if (pivotRow = ...)
  break;
  
is cleaner and then you can skip the continue